flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/tableImpl.scala

```diff
@@ -564,26 +575,47 @@ class WindowGroupedTableImpl(
       fields)
   }
 
+  private def expandColumnsInGroupWindow(window: GroupWindow): GroupWindow = {
```

Comment:
Why do we need it?

Reply:
The time filed in GroupWindow can also contains a column operation. For example, 
```
      .window(Slide.over("3.milli").every("10.milli").on("columns(b)").as("w"))
```
And time field are passed to the `extractFieldReferences()`:
```
    val projectFields = extractFieldReferences(
      (expressionsWithResolvedCalls ++ resolvedGroupKeys :+ resolvedWindow.getTimeField))
```
if we don't expand column operations for time field, `extractFieldReferences` would be failed.

Reply:
I think we should rather modify the `extractFieldReference` to extract also column expressions (They are actually field references). That way we will have a more unified approach and we can use the rule to expand them only in the `ExpressionResolver`

Reply:
This would bring some duplicate code. However, it would also make sense. I will try this.

Reply:
But the code in `extractFieldReference` would be super simple. I haven't checked it, but this should work:

```
		@Override
		public List<Expression> visitCall(CallExpression call) {
			FunctionDefinition functionDefinition = call.getFunctionDefinition();
			if (WINDOW_PROPERTIES.contains(functionDefinition)) {
				return Collections.emptyList();
			} else if (COLUMN_OPERATIONS.contains(functionDefinition)) {
				return call;
			} else {
				return call.getChildren()
					.stream()
					.flatMap(c -> c.accept(this).stream())
					.distinct()
					.collect(Collectors.toList());
			}
		}
```

Reply:
We can't return the column expression directly. We need to expand it as all references need to be distincted.
```
	public static List<Expression> extractFieldReferences(List<Expression> expressions) {
		return expressions.stream()
			.flatMap(expr -> expr.accept(referenceExtractor).stream())
			.distinct()
			.collect(Collectors.toList());
	}
```